cloudinit/dmi.py

```diff
@@ -111,6 +140,9 @@ def read_dmi_data(key):
     if syspath_value is not None:
         return syspath_value
 
+    if is_FreeBSD():
+        return _read_kenv(key)
+
```

Comment:
seems like you might as well bump this up above the '_read_dmi_syspath' call.
theres no point in looking in /sys on freebsd, right? So that would always be a wasted stat (inside _read_dmi_syspath) on freebsd.  And on linux we'd just pay the '@lru_cache' call is_FreeBSD.

I'm not sure if that is clear...
short summary: just move this up above the call to _read_dmi_syspath.